matrix/Quaternion.hpp

```diff
@@ -195,9 +195,10 @@ class Quaternion : public Vector<Type, 4>
         Quaternion &q = *this;
         Vector3<Type> cr = src.cross(dst);
         float dt = src.dot(dst);
-        /* If the two vectors are parallel, cross product is zero
-         * If they point opposite, the dot product is negative */
         if (cr.norm() < eps && dt < 0) {
+            // handle corner cases with 180 degree rotations
+            // if the two vectors are parallel, cross product is zero
+            // if they point opposite, the dot product is negative
             cr = src.abs();
```

Comment:
No because in this corner case `cr` the cross product was a zero vector which is useless to determine the rotation direction. So we infer the adequate direction by a fused method, more details in the original description: https://github.com/PX4/Matrix/pull/55#issue-152273286 For that cr needs to contain something different.

Reply:
It's incredibly minor, but I was only suggesting not reusing variables (cr) for different purposes. 

If I didn't miss anything (on a tiny screen) cr is being used as a temp to hold the source absolute values, then a unit vector, then finally the cross product of source with that unit vector.